pkgs/development/tools/vcpkg/default.nix

```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, stdenv
+, lib
+, vcpkg-tool
+, writeShellScript
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vcpkg";
+  version = "2023.08.09";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "vcpkg";
+    rev = finalAttrs.version;
+    hash = "sha256-EK8Cikxn4yYvU6/xofNCUBEdJeGMNuoVRHHq+3sFYeA=";
+  };
+
+  installPhase = let
+    # vcpkg needs two directories to write to that is independent of installation directory.
+    # Since vcpkg already creates $HOME/.vcpkg/ we use that to create a root where vcpkg can write into.
+    vcpkgScript = writeShellScript "vcpkg" ''
+      vcpkg_writable_path="$HOME/.vcpkg/root/"
+
+      VCPKG_ROOT="$out/share/vcpkg" ${vcpkg-tool}/bin/vcpkg \
+        --x-downloads-root=$vcpkg_writable_path/downloads \
+        --x-buildtrees-root=$vcpkg_writable_path/buildtrees \
+        --x-packages-root=$vcpkg_writable_path/packages \
+        "$@"
+      '';
+    in ''
+      mkdir -p $out/bin $out/share/vcpkg/scripts/buildsystems
+      cp --preserve=mode -r ./{docs,ports,triplets,scripts,.vcpkg-root,versions,LICENSE.txt} $out/share/vcpkg/
+      sed "s:\''$out:$out:" ${vcpkgScript} > $out/bin/vcpkg
```

Comment:
I'm not very proud of this code right there, but it actually works. I know there must be a better way. Can someone with a better knowledge of nix lang help me making this code less horrible? Thanks.

Reply:
I have a feeling you should try `substituteAll`.